core/products/perpetual.go

```diff
@@ -582,17 +594,16 @@ func (p *Perpetual) UpdateAuctionState(ctx context.Context, enter bool) {
 
 	// left first auction, we can start the first funding-period
 	p.startedAt = t
+	// store internal data submitted at the end of opening auction
+	temp := p.internalTWAP
 	p.internalTWAP = NewCachedTWAP(p.log, t, p.auctions)
-	p.externalTWAP = NewCachedTWAP(p.log, t, p.auctions)
-	p.broker.Send(events.NewFundingPeriodEvent(ctx, p.id, p.seq, p.startedAt, nil, nil, nil, nil, nil))
+	p.internalTWAP.sumProduct = temp.sumProduct.Clone()
+	p.internalTWAP.points = temp.points
```

Comment:
I don't think we can carry across the cached `sumProduct` like this because we're changing the start of the TWAP period so it'll be stale. Can you instead just repopulate the new cached-twap by calling `addPoint()` with the points from the old one?

Or alternatively reorder the calls in `leaveAuction()` in `market.go` so that we call `SubmitDataPoint()` after we call `UpdateAuctionState()`, then we can avoid having to fiddle like this.